core/base-service/coalesce-badge.spec.js

```diff
@@ -0,0 +1,257 @@
+      ).to.deep.equal(['purr count', 'n/a'])
+    })
+  })
+
+  describe('Message', function() {
+    it('applies the service message', function() {
+      expect(coalesceBadge({}, { message: '10k' }, {}).text).to.deep.equal([
+        undefined,
+        '10k',
+      ])
+    })
+
+    it('applies a numeric service message', function() {
+      // While a number of badges use this, in the long run we may want
+      // `render()` to always return a string.
+      expect(coalesceBadge({}, { message: 10 }, {}).text).to.deep.equal([
+        undefined,
+        10,
+      ])
+    })
+  })
+
+  describe('Right color', function() {
+    it('uses the default color', function() {
+      expect(coalesceBadge({}, {}, {}).color).to.equal('lightgrey')
+    })
+
+    it('overrides the color', function() {
+      expect(
+        coalesceBadge({ colorB: '10ADED' }, { color: 'red' }, {}).color
+      ).to.equal('10ADED')
+    })
+
+    context('In case of an error', function() {
+      it('does not override the color', function() {
+        expect(
+          coalesceBadge(
+            { colorB: '10ADED' },
+            { isError: true, color: 'lightgray' },
+            {}
+          ).color
+        ).to.equal('lightgray')
+      })
+    })
+
+    it('applies the service color', function() {
+      expect(coalesceBadge({}, { color: 'red' }, {}).color).to.equal('red')
+    })
+  })
+
+  describe('Left color', function() {
+    it('provides no default label color', function() {
+      expect(coalesceBadge({}, {}, {}).labelColor).to.be.undefined
+    })
+
+    it('applies the service label color', function() {
+      expect(coalesceBadge({}, { labelColor: 'red' }, {}).labelColor).to.equal(
+        'red'
+      )
+    })
+
+    it('overrides the label color', function() {
+      expect(
+        coalesceBadge({ colorA: '42f483' }, { color: 'green' }, {}).labelColor
+      ).to.equal('42f483')
+    })
+
+    it('converts a query-string numeric color to a string', function() {
+      expect(
+        coalesceBadge(
+          // Scoutcamp converts numeric query params to numbers.
+          { colorB: 123 },
+          { color: 'green' },
+          {}
+        ).color
+      ).to.equal('123')
+    })
+  })
+
+  describe('Named logos', function() {
+    it('when not a social badge, ignores the default named logo', function() {
+      expect(coalesceBadge({}, {}, { namedLogo: 'appveyor' }).logo).to.be
+        .undefined
+    })
+
+    it('when a social badge, uses the default named logo', function() {
+      // .not.be.empty for confidence that nothing has changed with `getShieldsIcon()`.
+      expect(
+        coalesceBadge({ style: 'social' }, {}, { namedLogo: 'appveyor' }).logo
+      ).to.equal(getShieldsIcon({ name: 'appveyor' })).and.not.be.empty
+    })
+
+    it('applies the named logo', function() {
+      expect(coalesceBadge({}, { namedLogo: 'github' }, {}).logo).to.equal(
+        getShieldsIcon({ name: 'github' })
```

Comment:
this will need to be switched to `getSimpleIcon` once #2857 is merged right?

Reply:
Heh yep. Or to `npm`.